tests/test_package_client_cpp.py

```diff
@@ -0,0 +1,254 @@
+    ):
+        result = setup_tester_ssh_connection.run("uname -a")
+        assert "raspberrypi" in result.stdout
+
+        # First things first
+        # Explicitly accept suite change from "stable" to "oldstable"
+        setup_tester_ssh_connection.run(
+            "sudo apt-get update --allow-releaseinfo-change-suite"
+        )
+
+        # Client meta package
+        upload_deb_package(
+            setup_tester_ssh_connection, mender_dist_packages_versions["mender-client"]
+        )
+        # Upload mender-auth
+        upload_deb_package(
+            setup_tester_ssh_connection,
+            mender_dist_packages_versions["mender-client"],
+            package_name="mender-auth",
+        )
+        # Upload mender-update
+        upload_deb_package(
+            setup_tester_ssh_connection,
+            mender_dist_packages_versions["mender-client"],
+            package_name="mender-update",
+        )
+
+        # Install the deb packages. On failure, install the missing dependencies.
+        result = setup_tester_ssh_connection.run(
+            "sudo apt install --yes ./"
+            + package_filename(mender_dist_packages_versions["mender-client"]).replace(
+                "client", "update"
+            )
+        )
+        assert (
+            "Unpacking mender-update ("
+            + mender_dist_packages_versions["mender-client"]
+            + ")"
+            in result.stdout
+        )
+        assert (
+            "Setting up mender-update ("
+            + mender_dist_packages_versions["mender-client"]
+            + ")"
+            in result.stdout
+        )
+
+        result = setup_tester_ssh_connection.run(
+            "sudo apt install --yes ./"
+            + package_filename(mender_dist_packages_versions["mender-client"]).replace(
+                "client", "auth"
+            )
+        )
+        assert (
+            "Unpacking mender-auth ("
+            + mender_dist_packages_versions["mender-client"]
+            + ")"
+            in result.stdout
+        )
+        assert (
+            "Setting up mender-auth ("
+            + mender_dist_packages_versions["mender-client"]
+            + ")"
+            in result.stdout
+        )
+
+        # Install the client meta-package also
+        result = setup_tester_ssh_connection.run(
+            "sudo apt install --yes ./"
+            + package_filename(mender_dist_packages_versions["mender-client"])
+        )
+
+        self.check_installed_files(setup_tester_ssh_connection, "raspberrypi")
+
+        # TODO - Cpp client does not yet report a version
+        # self.check_mender_client_version(setup_tester_ssh_connection, mender_version)
+
+        # Default setup expects ServerURL hosted.mender.io
+        # TODO - No mender-setup yet
+        # result = setup_tester_ssh_connection.sudo("cat /etc/mender/mender.conf")
+        # assert '"ServerURL": "https://hosted.mender.io"' in result.stdout
```

Comment:
This is now tested separately on the `mender-setup` tests. This block can be removed